packages/flutter/lib/src/painting/image_cache.dart

```diff
@@ -47,22 +49,69 @@ class ImageCache {
     _maximumSize = value;
     if (maximumSize == 0) {
       _cache.clear();
+      _sizeBytes = 0;
     } else {
-      while (_cache.length > maximumSize)
-        _cache.remove(_cache.keys.first);
+      _checkCacheSize();
     }
   }
 
+  /// The current number of cached entries.
+  int get currentSize => _cache.length;
+
+  /// Maximum size of entries to store in the cache in bytes.
+  ///
+  /// Once more than this amount of bytes have been cached, the
+  /// least-recently-used entry is evicted until there are fewer than the
+  /// maximum bytes.
+  int get maximumSizeBytes => _maximumSize;
```

Comment:
_maximumSizeBytes, not _maximumSize.

Please make sure to add a test that would have caught this.

Reply:
Should be covered by tests which assert cache size is respected